tools/cmake/mbed_greentea.cmake

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) 2020 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0# the macro
+
+set(MBED_CONFIG_PATH ${CMAKE_CURRENT_SOURCE_DIR}/.mbedbuild CACHE INTERNAL "")
+
+include(${MBED_PATH}/tools/cmake/app.cmake)
+
+# CMake Macro for generalizing CMake configuration across the greentea test suite with configurable parameters
+# Macro args:
+# TEST_NAME - Test suite name
+# TEST_INCLUDE_DIRS - Test suite include directories for the test
+# TEST_SOURCES - Test suite sources
+# TEST_REQUIRED_LIBS - Test suite required libraries
+# 
+# calling the macro:
+# mbed_greentea_cmake_macro(TEST_NAME mbed-platform-system-reset TEST_INCLUDE_DIRS mbed_store TEST_SOURCES foo.cpp bar.cpp TEST_REQUIRED_LIBS mbed-kvstore mbed-xyz)
+
+macro(mbed_greentea_cmake_macro)
```

Comment:
Can we name this to something that describes what it's actually doing? The name `mbed_greentea_cmake_macro` isn't really helping me understand why I would use this; all I can infer from the name is it's a cmake macro somehow related to greentea. It looks like it's adding a test executable target, so maybe `mbed_greentea_add_test` is a better name?

Reply:
I agree!

Reply:
I also agree but to be more efficient I suggest it stays as is. The reason being that there are multiple PRs that rely on this PR and have called the macro with the current name. It will be more efficient to merge all PRs and then do a search and replace than having to go and modify all these other PRs.
The search and replace can be done as soon as the other PRs are merged then a single PR to amend the name can be raised.

Reply:
`mbed_greentea_add_test ` would be better.

This is similar to what apps need to define (we do not provide a macro for the application but here for tests). My question would be shall we also have this or what is the drive to have it for tests?

Reply:
@0xc0170 https://github.com/ARMmbed/mbed-os/pull/14072#issuecomment-749022939